engine/src/test/java/com/datatorrent/stram/StramMiniClusterTest.java

```diff
@@ -165,22 +168,27 @@ public static void tearDown() throws IOException
     }
   }
 
-  @Test
-  public void testSetupShutdown() throws Exception
+  private void checkNodeState() throws YarnException
   {
-    GetClusterNodesRequest request =
-        Records.newRecord(GetClusterNodesRequest.class);
+    GetClusterNodesRequest request = Records.newRecord(GetClusterNodesRequest.class);
     ClientRMService clientRMService = yarnCluster.getResourceManager().getClientRMService();
     GetClusterNodesResponse response = clientRMService.getClusterNodes(request);
     List<NodeReport> nodeReports = response.getNodeReports();
     LOG.info("{}", nodeReports);
 
     for (NodeReport nr: nodeReports) {
-      LOG.info("Node: {}", nr.getNodeId());
-      LOG.info("Total memory: {}", nr.getCapability());
-      LOG.info("Used memory: {}", nr.getUsed());
-      LOG.info("Number containers: {}", nr.getNumContainers());
+      if (nr.getNodeState().isUnusable()) {
```

Comment:
couldn't you simplify this with if (!nr.getNodeState().isUnusable()) return; and not do anything when it is unusable as it will continue the for loop anyway.